ironfish/src/network/peers/peerManager.ts

```diff
@@ -866,6 +876,8 @@ export class PeerManager {
       }
       this.peers = this.peers.filter((p) => p !== peer)
 
+      this.addressManager.removePeerAddress(peer)
```

Comment:
This is a slightly tighter bound than we need to remove a peer from the peer addresses -- really we only care if the connection retry is in `willNeverRetryConnecting`, regardless of whether the peer has a connected peer.

Thinking about this also made me again consider whether `possiblePeers` is useful to keep around. I think if a peer address gets removed, it'll get added back into the possiblePeers next time a peerlist message is received. In other words, I think right now, on average `possiblePeers` will contain every peer with an address/port in `this.peers`.

Reply:
I'll adjust the bound. As for `possiblePeers`, I'm inclined to agree with you; I think it requires a fair bit of work to ensure that it functions as intended.